### ISSUE TITLE
Compiling C and C++ files together

### ISSUE THREAD

**Issue by BdT141** [Alex079/vscode-avr-helper#13]
Hi, 

I'm struggling to get a project compiled that has C and C++ files mixed in together. The C++ files compile fine, but it fails when it tries to compile the C files. I get the following output:

===== Compiling .cpp files
===== Making .vscode/avr.build/obj/main.cpp.o

avr-g++.exe -DF_CPU=16000000UL -mmcu=atmega328p -pipe  -std=c++14 -g -Os -Wall -Wextra -pedantic -c -fpermissive -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -MMD -flto main.cpp -o .vscode/avr.build/obj/main.cpp.o
===== Compiling .C files
===== Making .vscode/avr.build/obj/twi.C.o

A subdirectory or file .vscode\avr.build\obj already exists.
make: [.vscode/avr.build/obj/twi.C.o] Error 1 (ignored)     
twi.C -o .vscode/avr.build/obj/twi.C.o
process_begin: CreateProcess(C:\Users\bdutoit\Desktop\AVR_TEST\twi.C, twi.C -o .vscode/avr.build/obj/twi.C.o, ...) failed.
make (e=193): Error 193
make: *** [.vscode/avr.build/obj/twi.C.o] Error 193
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command make -fc:\Users\bdutoit\Desktop\AVR_TEST\.vscode\avr.targets.mk build" terminated with exit code: 1.

When I compile the same project in Atmel Studio I see it compiles the C++ files with avr-g++ and the C files with avr-gcc. Do you support differentiating compilers based to file types? Is this even an issue?

**Comment by Alex079** [comment]
Hi!

I added _mkdir_ in every obj target because similarly named source files can potentially be located in different subfolders.
You can prepend a `mkdir....` command with a `-` symbol to let the build pass even if _mkdir_ fails.
The compiler is supposed to accept all supported types. Mixing C and C++ should not be an issue.
But Windows is case-insensitive, I will fix it.

Thanks for your help!

**Comment by BdT141** [comment]
I think I found the cause, but I don't know if the solution is build-related, compile-related or language-related.

When coding in Atmel Studio I had to do an `extern "C"{ #include "twi.h" }` for the compiler to find the C functions. Using your extension I need to explicitly NOT do the `extern "C"` when including C files for them to compile correctly.

Also I have attached the avr.targets.mk file, as a text file, that works for me on Windows, if it helps.
[avr.targets.txt](https://github.com/Alex079/vscode-avr-helper/files/6425704/avr.targets.txt)
